src/lib/util/params.tsx

```diff
@@ -0,0 +1,51 @@
+type Param = string | string[] | undefined;
+
+export function parseStringParam(param: Param): string | undefined;
+export function parseStringParam<T>(param: Param, defaultVal: T): string | T;
+export function parseStringParam(
```

Comment:
😕Sry... I'm not super familiar with Typescript yet. What's happening here with all the exports? Can you point me at some documentation?

Reply:
These are overloaded function signatures. (https://www.typescriptlang.org/docs/handbook/functions.html#overloads)

It lets us define a different return type based on which arguments are provided.

`parseStringParam` can be called with or without a `defaultVal` argument. When it is called without a default value the function returns either a string or undefined. When a default value is provided, it returns either a string or that value's type.

The overloaded signature with the generic `<T>` allows TypeScript to forward the type of the default value arg to the return type.

It would be nicer if we could instead type the function without an overload, but TypeScript doesn't allow an argument that defines a generic type to be optional.

example: https://www.typescriptlang.org/play/#src=%2F%2F%20This%20doesn't%20work%20%3A(%0D%0Adeclare%20function%20func1%3CT%20%3D%20undefined%3E(arg%3A%20T%20%3D%20undefined)%3A%20string%20%7C%20T%3B%0D%0A%0D%0A%0D%0A%2F%2F%20But%20this%20does%20%3AD%0D%0Adeclare%20function%20func2()%3A%20string%20%7C%20undefined%3B%0D%0Adeclare%20function%20func2%3CT%3E(arg%3A%20T)%3A%20string%20%7C%20T%3B%0D%0A

Reply:
😮 Thanks for the explainer!